gr/include/gnuradio/port.h

```diff
@@ -80,10 +80,10 @@ class GR_RUNTIME_API port_base : public port_interface,
     void set_parent_intf(neighbor_interface_sptr intf) { _parent_intf = intf; }
     std::string format_descriptor();
     void set_format_descriptor(const std::string& fd) { _format_descriptor = fd; }
-    void set_buffer(buffer_sptr buffer) { _buffer = buffer; }
-    buffer_sptr buffer() { return _buffer; }
-    void set_buffer_reader(buffer_reader_sptr rdr) { _buffer_reader = rdr; }
-    buffer_reader_sptr buffer_reader() { return _buffer_reader; }
+    void set_buffer(buffer_uptr& buffer) { _buffer = std::move(buffer); }
```

Comment:
Use `&&`, otherwise callers can call it with an lvalue and be surprised that it's modified.